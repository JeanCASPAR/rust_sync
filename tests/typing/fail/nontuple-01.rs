use rustre::sync;

sync! {
    #![pass(1)]

    node non(a : int, b : int) = (a, b);

    node oui() = ()
    where
        (a : int, b : int) = if true {
            non(1, 2)
        } else {
            non(2, 1)
        };
}

fn main() {}
