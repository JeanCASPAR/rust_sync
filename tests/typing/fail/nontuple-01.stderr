error: tuples don't exist: int * int is not a valid type...
 --> tests/typing/fail/nontuple-01.rs:9:13
  |
9 |             non(1, 2)
  |             ^^^^^^^^^
